src/runtime/auto-prompt-manager.ts

```diff
@@ -467,6 +504,110 @@ export class AutoPromptManager {
     return potentialAction;
   }
 
+  private async getTargetedInterventionOrchestration(
+    clientConfig: ClientConfig,
+    article: Article,
+    contentType: ContentType
+  ): Promise<InterventionOrchestration | void> {
+    let eligibleActions = article.audienceActions?.actions;
+    let targetedInterventions =
+      article.actionOrchestration?.interventionFunnel?.prompts;
+    if (
+      !eligibleActions ||
+      eligibleActions.length === 0 ||
+      !targetedInterventions ||
+      targetedInterventions.length === 0
+    ) {
+      return;
+    }
+
+    // Complete client-side eligibility checks for actions.
+    const actionsTimestamps = await this.getTimestamps();
+    eligibleActions = eligibleActions.filter((action) =>
+      this.checkActionEligibility_(action.type, actionsTimestamps!)
+    );
+    if (eligibleActions.length === 0) {
+      return;
+    }
+
+    // Filter the funnel of interventions by eligibility.
+    targetedInterventions = targetedInterventions.filter((intervention) =>
+      eligibleActions
```

Comment:
nit: looks like it's dealing with array conversion every iteration within outer filter loop. What about creating a set of eligible config ids instead?